# ASSIGNMENT :-4 (CHI-SQUARE TEST)

Association between Device Type and Customer Satisfaction

##  State the Hypotheses :

Null Hypothesis (H0):
There is no association between device type (Smart Thermostat vs. Smart Light) and customer satisfaction level. Customer satisfaction is independent of device type.

Alternative Hypothesis (H1):
There is an association between device type and customer satisfaction level. Customer satisfaction depends on device type.

## Data Provided (Contingency Table)

In [10]:
import numpy as np
import pandas as pd 
from scipy.stats import chi2
from scipy.stats import chi2_contingency
from scipy.stats import chi2 as chi2_dist  

In [11]:
# Data

data = np.array([
    [50, 70],
    [80, 100],
    [60, 90],
    [30, 50],
    [20, 50]
])
data

array([[ 50,  70],
       [ 80, 100],
       [ 60,  90],
       [ 30,  50],
       [ 20,  50]])

In [12]:
# Convert to DataFrame

df = pd.DataFrame(data, 
                  index=["Very Satisfied", "Satisfied", "Neutral", "Unsatisfied", "Very Unsatisfied"], 
                  columns=["Smart Thermostat", "Smart Light"])
df

Smart Thermostat  Smart Light
Very Satisfied                  50           70
Satisfied                       80          100
Neutral                         60           90
Unsatisfied                     30           50
Very Unsatisfied                20           50

## Compute the Chi-Square Statistic 

In [13]:
# Chi-Square Test

chi2_stat, p, dof, expected = chi2_contingency(df)

print("\nChi-Square Statistic:", chi2_stat)
print("Degrees of Freedom:", dof)
print("P-value:", p)
print("Expected Frequencies:\n", expected)


Chi-Square Statistic: 5.638227513227513
Degrees of Freedom: 4
P-value: 0.22784371130697179
Expected Frequencies:
 [[ 48.  72.]
 [ 72. 108.]
 [ 60.  90.]
 [ 32.  48.]
 [ 28.  42.]]


In [14]:
#show expected frequencies
expected_df = pd.DataFrame(expected, index=df.index, columns=df.columns)
expected_df

Smart Thermostat  Smart Light
Very Satisfied                48.0         72.0
Satisfied                     72.0        108.0
Neutral                       60.0         90.0
Unsatisfied                   32.0         48.0
Very Unsatisfied              28.0         42.0

##  Determine the Critical Value

In [15]:
alpha = 0.05
alpha

0.05

In [16]:
critical_value = chi2.ppf(1 - alpha, dof)
critical_value

9.487729036781154

##  Make a Decision

In [18]:
if chi2_stat > critical_value:
    print("\nDecision: Reject Null Hypothesis (Significant association)")
else:
    print("\nDecision: Fail to Reject Null Hypothesis (No significant association)")


Decision: Fail to Reject Null Hypothesis (No significant association)


## Conclusion

At the 5% significance level, there is no significant association between the type of smart home device (Smart Thermostat vs. Smart Light) and customer satisfaction level.

This means that customer satisfaction appears to be independent of device type.

# ASSIGNMENT :- 4 (HYPOTHESIS TESTING)

### Data Provided: 

The theoretical weekly operating cost model: W = $1,000 + $5X

Sample of 25 restaurants with a mean weekly cost of Rs. 3,050

Number of units produced in a week (X) follows a normal distribution with a mean (μ) of 600 units and a standard deviation (σ) of 25 units

ˉxˉ = sample mean weekly cost (Rs. 3,050)

μ = theoretical mean weekly cost according to the cost model (W = $1,000 + $5X for X = 600 units)

σ = 5*25 units

n = sample size (25 restaurants)

### 1. State the Hypotheses statement

restaurant owners claim weekly operating cost are higher than predicted by the mode

Null hypothesis (H0) : the average weekly cost of restaurants is the same as predicted by model  

Alternative hypotesis(H1) : the average weekly operating cost of the restaurants is higher than what the model prdicts 

Null hypothesis (H0) : μ = 4000

Alternative hypotesis (H1) : μ > 4000

### 2. Calculate the Test Statistic

In [20]:
import math 
from scipy.stats import norm 

In [21]:
mean=3050   # sample mean 
mean

3050

In [22]:
m_units=600   # mean units 
m_units

600

In [23]:
std= 25   # standard deviation of units
std

25

In [24]:
std_cost = 5 * std    # unit production 
std_cost

125

In [25]:
n = 25
n

25

In [26]:
μ = 1000 + 5 * m_units   # theoretical mean 
μ

4000

In [27]:
SE = std_cost / math.sqrt(n)   # standard error
SE

25.0

In [28]:
z = (mean - μ) / SE   # Test statistics ( population standard deviation is known , so use z)
z

-38.0

#### Conclusions :
we calculate z = -38.0 
the test ststistic is extreamly negative . This show that sample mean (3050) is much smaller than theoretical mean (4000).
The test statistic provide no support for the claim that actual costs are higher instead , it suggest they are significantly lower

### 3. Determine the Critical Value

In [29]:
alpha = 0.05   # given 
alpha

0.05

In [30]:
z_critical = norm.ppf(1- alpha)
z_critical

1.6448536269514722

### Conclusions:
if z > critical value then H0 reject . since z (-38) does not exceed the critical value (1.645), we do not reject H0.

### 4. Make a Decision

In [31]:
p_value = 1 - norm.cdf(z)
p_value

1.0

In [32]:
if z > z_critical:
    print(" Decision : Reject H0 (evidence costs is heigher).")
else:
    print(" Decision : Do Not Reject H0 (no evidence costs is heigher).")

 Decision : Do Not Reject H0 (no evidence costs is heigher).


### conclusion:
p_value is 1.0 , much lrger than 0.05 . The sample provide that weekly costs are higher than model predicts . The cost are lower

## 5. Conclusion:
There is no evidence to support the restaurant owners' claim that the weekly operating costs are higher than the model suggests. The observed mean is (3050) is much lower than the model mean (4000) and the test statistic z = -38 strongly contradicts the claim that costs increased.